app/javascript/src/javascripts/uploader/file_preview.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <div class="upload_preview_container" :class="classes">
+    <div class="box-section sect_red" v-show="overDims">
+      One of the image dimensions is above the maximum allowed of 15,000px and will fail to upload.
+    </div>
+    <div v-if="!failed">
+      <div class="upload_preview_dims">{{ previewDimensions }}</div>
+      <preview-video v-if="isVideo" :url="finalPreviewUrl"
```

Comment:
Due to the tightly coupled nature of the isVideo and url variables, and the extreme similarity of the way the preview is displayed. I think it makes more sense to package these two variables up into an object that is emitted and then consumed by a single component, which internally makes the decision to display a video or image tag.

By having two variables you may run into subtle race conditions(it seems unlikely, but it isn't guaranteed not to happen), where passing around a new instance of an object each time would result in a single prop that changes instead of the internal members. This will always arrive as a single piece and be processed in the same render tick.

Reply:
I didn't know that, good to know. I added a validator to check the objects shape. Also makes it more explicit in the code what properties are usable/needed.